src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/CommandProcessor.java

```diff
@@ -599,32 +599,8 @@ public void doit(Tokens t) {
                 if (t.countTokens() != 1) {
                     usage();
                 } else {
-                    String symbol = t.nextToken();
-                    Address addr = VM.getVM().getDebugger().lookup(null, symbol);
-                    if (addr == null && VM.getVM().getDebugger().getOS().equals("win32")) {
-                        // On win32 symbols are prefixed with the dll name. Do the user
-                        // a favor and see if this is a symbol in jvm.dll or java.dll.
-                        addr = VM.getVM().getDebugger().lookup(null, "jvm!" + symbol);
-                        if (addr == null) {
-                            addr = VM.getVM().getDebugger().lookup(null, "java!" + symbol);
-                        }
-                    }
-                    if (addr == null) {
-                        out.println("Symbol not found");
-                        return;
-                    }
-                    out.print(addr);  // Print the address of the symbol.
-                    CDebugger cdbg = VM.getVM().getDebugger().getCDebugger();
-                    LoadObject loadObject = cdbg.loadObjectContainingPC(addr);
-                    // Print the shared library path and the offset of the symbol.
-                    if (loadObject != null) {
-                        out.print(": " + loadObject.getName());
-                        long diff = addr.minus(loadObject.getBase());
-                        if (diff != 0L) {
-                            out.print(" + 0x" + Long.toHexString(diff));
-                        }
-                    }
-                    out.println();
+                    String result = VM.getVM().getDebugger().findSymbol(t.nextToken());
+                    out.println(result == null ? "Symbol not found" : result);
```

Comment:
The import of sun.jvm.hotspot.debugger.cdbg.CDebugger is no longer needed in this file.

I think the more traditional approach would be to add remote support for LoadObject. That provides the building blocks needed here. It's harder to do, but might have some benefit down the road, although it's hard to say if there would ever be other users of it.

Reply:
I removed both CDebugger and LoadObject from CommandProcessor.java in new commit.

> I think the more traditional approach would be to add remote support for LoadObject. That provides the building blocks needed here. It's harder to do, but might have some benefit down the road, although it's hard to say if there would ever be other users of it.

Agree. I saw similar issue in `dis`, `disassemble`, `dumpcodecache`, and `livenmethods`. I hope they will be fixed, and I believe remote support for LoadObject helps it. (They can be fixed in Disassembler.java)

I think LoadObject should be Serializable if we do so, but it might make big impact for the classes which implements LoadObject (e.g. DSO.java). It might be harder and we might have no more users for it as you said, so I decided to delegate the process to remote debug server.